src/Nethermind/Nethermind.Blockchain/Find/LogFinder.cs

```diff
@@ -63,18 +63,29 @@ public class LogFinder : ILogFinder
         public IEnumerable<FilterLog> FindLogs(LogFilter filter, CancellationToken cancellationToken = default)
         {
             BlockHeader FindHeader(BlockParameter blockParameter, string name, bool headLimit) => 
-                _blockFinder.FindHeader(blockParameter, headLimit) ?? throw new ArgumentException("Block not found.", name);
+                _blockFinder.FindHeader(blockParameter, headLimit) ?? throw new ResourceNotFoundException($"Block not found: {name}");
```

Comment:
Can we also include BlockParameter in error message?

Reply:
Seems to already do it via the `name` parameter.

Reply:
No its just nameof(filter.ToBlock), lets add what was that what we are missing (blocknumber, hash)

Reply:
Ok, I suppose a simple `.ToString()` is enough?

Reply:
yes just drop it somewhere in the pattern in some good way